code/datums/components/infective.dm

```diff
@@ -58,7 +58,8 @@
 
 /datum/component/infective/proc/try_infect_eat(datum/source, mob/living/eater, mob/living/feeder)
 	SIGNAL_HANDLER
-
+	if(isflyperson(eater))
+		return // flies can eat food off the floor just fine
```

Comment:
Can you make this check for GROSS food preference instead? You can swap out the deviant tastes check aswell then

Reply:
But deviants aren't immune to food illnesses? They just don't get the mood debuff

Reply:
Oh, I read that wrong. Still doesn’t make sense though, that should at least abide by tasta